File: Selection.php
<?php
function selectionSort(&$numeros) {
    $n = count($numeros);
    for ($i = 0; $i < $n - 1; $i++) {
        $menorIndice = $i;
        for ($j = $i + 1; $j < $n; $j++) {
            if ($numeros[$j] < $numeros[$menorIndice]) {
                $menorIndice = $j;
            }
        }
        $temp = $numeros[$i];
        $numeros[$i] = $numeros[$menorIndice];
        $numeros[$menorIndice] = $temp;
    }
}

function lerArquivo($caminho) {
    $conteudo = file($caminho, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    return array_map('intval', $conteudo);
}

function arquivo_ordenado($caminho, $numeros) {
    file_put_contents($caminho, implode(PHP_EOL, $numeros));
}

function obterInfoProcessador() {
    return shell_exec('lscpu');
}


echo "Linguagem: PHP " . PHP_VERSION . "\n";
echo "Sistema: " . PHP_OS . " - " . php_uname('r') . "\n";

echo "Processador:\n" . obterInfoProcessador() . "\n";

$memoriaTotalKB = trim(shell_exec("grep MemTotal /proc/meminfo | awk '{print $2}'"));
$memoriaTotalGB = round($memoriaTotalKB / 1024 / 1024, 2);
echo "Memória RAM total: " . $memoriaTotalGB . " GB\n";


$arquivoEntrada = 'arq-desafio.txt';
$arquivoSaida = 'ordenado_selection_php.txt';

$inicioTempo = microtime(true);
$memoriaInicial = memory_get_usage() / 1024;

$numeros = lerArquivo($arquivoEntrada);
selectionSort($numeros);
arquivo_ordenado($arquivoSaida, $numeros);

$fimTempo = microtime(true);
$memoriaFinal = memory_get_usage() / 1024;

echo "Números ordenados salvos em '$arquivoSaida'.\n";
echo "Tempo de execução: " . round(($fimTempo - $inicioTempo) * 1000, 2) . " ms\n";
echo "Memória utilizada: " . round($memoriaFinal - $memoriaInicial, 2) . " KB\n";
?>

File: Selection.py
import time
import psutil
import platform
import cpuinfo
import subprocess

def selection_sort(numeros):
    n = len(numeros)
    for i in range(n):
        menor_indice = i
        for j in range(i + 1, n):
            if numeros[j] < numeros[menor_indice]:
                menor_indice = j
        numeros[i], numeros[menor_indice] = numeros[menor_indice], numeros[i]

def ler_arquivo(caminho):
    with open(caminho, 'r', encoding='utf-8-sig') as arquivo:
        return [int(linha.strip()) for linha in arquivo]

def arquivo_ordenado(caminho, numeros):
    with open(caminho, 'w') as arquivo:
        arquivo.writelines(f"{num}\n" for num in numeros)

def obter_info_lscpu():

        cpu_info = subprocess.check_output("lscpu", shell=True, text=True)
        return cpu_info

print(f"Linguagem: Python {platform.python_version()}")

info_sistema = platform.uname()
print(f"Sistema: {info_sistema.system} {info_sistema.release}")

print(f"Processador (CPU info): {cpuinfo.get_cpu_info().get('brand_raw', 'Informação indisponível')}")

print("processador:")
print(obter_info_lscpu())

print(f"Memória RAM total: {psutil.virtual_memory().total / (1024 ** 3):.2f} GB")  

entrada = 'arq-desafio.txt'
saida = 'ordenado_seletion_py.txt'

inicio_tempo = time.time()
processo = psutil.Process()

mem_inicio = processo.memory_info().rss / 1024
numeros = ler_arquivo(entrada)
selection_sort(numeros)
arquivo_ordenado(saida, numeros)
mem_fim = processo.memory_info().rss / 1024

fim_tempo = time.time()

print(f"Números ordenados foram salvos em '{saida}'.")
print(f"Tempo de execução: {round((fim_tempo - inicio_tempo) * 1000, 2)} ms")
print(f"Memória utilizada: {round(mem_fim - mem_inicio, 2)} KB")
